### PR TITLE
Schema extensions

### PR DESCRIPTION
**Pull request by MIchaelMainer**
Forked off of OneNote work so only review commit 2b377e4. While I've seen the test run fine, we are blocked on running the tests again with this app registration since I can't delete the existing schema extensions and we've hit the max for number of schema extensions per app. The state machine appears broken so I can't get an extension to a deletable state.

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -115,3 +115,46 @@ Expand, Skip, Top, OrderBy, and Filter are also supported via the client library
 ## Collections
 
 Please see [collections](/docs/collections.md) for details on collections and paging.
+
+## Use the .Net Microsoft Graph client library when the client doesn't support a feature
```
Comment (dan-silver) [inline docs/overview.md]: I think this is a bit confusing if someone isn't familiar with the library :) Use the library when the library doesn't support x.  Maybe rephrase to something like "sending generic requests with the client library"?

File: docs/overview.md [hunk 2]
```diff
@@ -115,3 +115,46 @@ Expand, Skip, Top, OrderBy, and Filter are also supported via the client library
...
+
+    // Get the OneNote page that we created.
+    if (response.IsSuccessStatusCode)
+    {
+        // Deserialize into OneNotePage object.
+        var content = await response.Content.ReadAsStringAsync();
+        OnenotePage page = graphClient.HttpProvider.Serializer.DeserializeObject<OnenotePage>(content);
+    }
+    else
+        throw new Exception("Error");
```
Comment (dan-silver) [inline docs/overview.md]: Is there a way to throw the actual error from the response object?

File: docs/overview.md (hunk 1 above)
Comment (MIchaelMainer) [inline docs/overview.md]: Good point. Will change.

File: docs/overview.md (hunk 2 above)
Comment (MIchaelMainer) [inline docs/overview.md]: Yes. Will fix.

**Comment by MIchaelMainer** [comment]
Fixes #132
